util-cgroups/src/hierarchy.rs

```diff
@@ -500,4 +500,75 @@ mod tests {
             "unexpected result {h:?}"
         );
     }
+
+    #[test]
+    fn test_cgroup_path_not_absolute() {
+        let path = PathBuf::from("./not_absolute_path");
+        let h = CgroupHierarchy::from_root_path(&path);
+        assert!(h.is_err());
+        let err = h.expect_err("Expected errror: HierarchyError::BadRoot(path)");
+        match err {
+            HierarchyError::BadRoot(path_buf) => {
+                assert_eq!(path, path_buf);
+            }
+            _ => assert!(false),
+        }
```

Comment:
In such cases, you should use `assert!( matches(!h, …))` to make the intent clear and readable.